proposals/low-level-struct-improvements.md

```diff
@@ -1,1734 +1 @@
-    b. Identify the escape scopes of all inputs to the method that are returnable (don't line up with `scoped` parameters)
-
-If any value in 2.b is smaller than 2.a then the method call must be illegal. Let's look at a few examples to illustrate the rules:
-
-```c#
-ref struct R { }
-
-class Program
-{
-    static void F0(ref R a, scoped ref R b) => throw null;
-
-    static void F1(ref R x, scoped R y)
-    {
-        F0(ref x, ref y);
-    }
-}
-```
-
-Looking at the call to `F0` lets go through (1) and (2). The parameters with potential for indirect return are `a` and `b` as both can be directly assigned. The arguments which line up to those parameters are:
-
-- `a` which maps to `x` that has *escape-scope* of *calling method*
-- `b` which maps to `y` that has with *escape-scope* of *current method*
-
-The set of returnable input to the method are
-
-- `x` with *escape-scope* of *calling method*
-- `ref x` with *escape-scope* of *calling method*
-- `y` with *escape-scope* of *current method*
-
-The value `ref y` is not returnable since it maps to a `scoped ref` hence it is not considered an input. But given that there is at least one input with a smaller *escape scope* (`y` argument) than one of the outputs (`x` argument) the method call is illegal. 
-
-A different variation is the following:
-
-```c#
-ref struct R { }
-
-class Program
-{
-    static void F0(ref R a, ref int b) => throw null;
-
-    static void F1(ref R x)
-    {
-        int y = 42;
-        F0(ref x, ref y);
-    }
-}
-```
-
-Again the parameters with potential for indirect return are `a` and `b` as both can be directly assigned. But `b` can be excluded because it does not point to a `ref struct` hence cannot be used to store `ref` state. Thus we have:
-
-- `a` which maps to `x` that has *escape-scope* of *calling method*
-
-The set of returnable input to the method are:
-
-- `x` with *escape-scope* of *calling method*
-- `ref x` with *escape-scope* of *calling method*
-- `ref y` with *escape-scope* of *current method*
-
-Given that there is at least one input with a smaller *escape scope* (`ref y` argument) than one of the outputs (`x` argument) the method call is illegal. 
-
-This is the logic that the method arguments must match rule is trying to encompass. It goes further as it considers both `scoped` as a way to remove inputs from consideration and `readonly` as a way to remove `ref` as an output (can't assign into a `readonly ref` so it can't be a source of output). These special cases do add complexity to the rules but it's done so for the benefit of the developer. The compiler seeks to remove all inputs and outputs it knows can't contribute to the result to give developers maximum flexibility when calling a member. Much like overload resolution it's worth the effort to make our rules more complex when it creates more flexibility for consumers.
+Document moved to https://github.com/dotnet/csharplang/tree/main/proposals/csharp-11.0/low-level-struct-improvements.md
```

Comment:
>Document moved [](http://example.com/codeflow?start=0&length=14)

Would the git history of the moved file be preserved if this commit is removed?

Reply:
No, sadly git loses track through moves.

Reply:
I'll do a merge instead of a squash, but I think git will still not provide a useful history through this change :-/